frappe/contacts/doctype/address/address.js

```diff
@@ -30,5 +30,13 @@ frappe.ui.form.on("Address", {
 				frappe.model.remove_from_locals(d.link_doctype, d.link_name);
 			});
 		}
+	},
+	after_save: function(frm) {
+		//redirect to link name
+		if(frm.doc.links[0]!=null){
```

Comment:
instead of redirecting directly to the parent,   set the `frappe.route_flag` whenever user click on `Add Address` then redirect t the parent document.

Also remove the else condition